src/librawspeed/common/Spline.h

```diff
@@ -0,0 +1,35 @@
+/*
+    RawSpeed - RAW file decoder.
+
+    Copyright (C) 2017 Robert Bieber
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+*/
+
+#pragma once
+
+#include "common/Common.h" // for ushort16
+#include "common/Point.h"  // for iPoint2D
+#include <vector>          // for vector
+
+namespace rawspeed {
+
+class Spline final {
+public:
+  static std::vector<ushort16>
+  calculateCurve(const std::vector<iPoint2D>& control_points);
```

Comment:
I.e.
```c++
class Spline final {
public:
  static std::vector<ushort16>
  calculateCurve(const std::vector<iPoint2D>& control_points) {
    const int num_coords = control_points.size();
    const int num_segments = num_coords - 1;
    // These are the constant factors for each segment of the curve.
...
```

Reply:
Ah, I see.  What's the guideline for putting implementations in .h files instead of a separate .cpp?

Reply:
Dunno, nothing specific.